zap/src/main/java/org/zaproxy/zap/scan/filters/ScanFilter.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2019 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.scan.filters;
+
+import org.zaproxy.zap.model.StructuralNode;
+
+/** @author KSASAN preetkaran20@gmail.com */
+public interface ScanFilter {
+
+    FilterResult isFiltered(StructuralNode node);
+
+    String getFilterType();
```

Comment:
Should i change it to Enum instead of String ?

Reply:
I have added this method but i not sure if this will be really beneficial, please provide your inputs.

Reply:
It's better with a `String` since we can add new filters any time.

I don't think it should be added to the interface, that's not needed/used by the actual filtering process (it's only used by an implementation, should be in `AbstractScanFilter` for example).

Reply:
hmm. correct. will do that.